### PR TITLE
Remove Yoda conditions and legacy code

### PR DESCRIPTION
**Pull request by 0dp**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | I have gone through `classes/Hooks.php` to change all yoda conditions. I have also removed `Hook::$_hook_modules_cache = $list;` and `static::$_hook_modules_cache_exec = $allHookRegistrations;` because they should've been removed in 1.6 (please correct me if I am wrong).
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | yes
| Fixed ticket?     | #26405 
| How to test?      | ~Please indicate how to best verify that this PR is correct.~
| Possible impacts? | ~Please indicate what parts of the software we need to check to make sure everything is alright.~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26404)

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
what's wrong with yoda conditions ? 😅

**Comment by 0dp** [comment]
Please advice on failed check

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I see no reason to remove yoda conditions :thinking:

**Comment by 0dp** [comment]
Any reason to keep them @PierreRambaud ?

**Comment by PierreRambaud** [comment]
> Any reason to keep them @PierreRambaud ?

It currently works and changing it brings no value.
I can return your own question, why remove Yoda conditions?

**Comment by 0dp** [comment]
> > Any reason to keep them @PierreRambaud ?
> 
> It currently works and changing it brings no value. I can return your own question, why remove Yoda conditions?

The same question is asked by @matks on the issue #26405 

Here you can see I argue why we should avoid them when they are not serving their intended purpose.

TLDR;
Readability and consistency

**Review by PrestaEdit — COMMENTED** [review]
Even if you change the condition, you can't delete this inside the same PR/commit.

File: classes/Hook.php [hunk 1]
```diff
@@ -523,9 +523,6 @@ public static function getHookModuleList()
         }
         Cache::store($cache_id, $list);
 
-        // @todo remove this in 1.6, we keep it in 1.5 for retrocompatibility
-        Hook::$_hook_modules_cache = $list;
```
Comment (PrestaEdit) [inline classes/Hook.php]: Even if you change the condition, you can't delete this inside the same PR/commit.

File: classes/Hook.php [hunk 2]
```diff
@@ -1208,8 +1207,6 @@ private static function getAllHookRegistrations(Context $context, ?string $hookN
 
         if ($useCache) {
             Cache::store($cache_id, $allHookRegistrations);
-            // @todo remove this in 1.6, we keep it in 1.5 for backward compatibility
-            static::$_hook_modules_cache_exec = $allHookRegistrations;
```
Comment (PrestaEdit) [inline classes/Hook.php:1212]: Even if you change the condition, you can't delete this inside the same PR/commit.

**Comment by PierreRambaud** [comment]
Hi @0dp 
Thanks for your contribution, but as we said in the issue, we will not accept your pull request without any good reason.
Kind regards
